environment/src/main/java/com/sequenceiq/environment/environment/service/freeipa/FreeIpaService.java

```diff
@@ -164,6 +164,30 @@ public Optional<OperationView> getFreeIpaOperation(String environmentCrn, boolea
         }
     }
 
+    public OperationStatus getOperationStatus(String operationId) {
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        try {
+            return operationV1Endpoint.getOperationStatus(operationId, accountId);
+        } catch (WebApplicationException e) {
+            String errorMessage = webApplicationExceptionMessageExtractor.getErrorMessage(e);
+            LOGGER.error("Failed to get operation status '{}' in account {} due to: '{}'", operationId, accountId, errorMessage, e);
+            throw new FreeIpaOperationFailedException(errorMessage, e);
+        }
+    }
+
+    public OperationStatus upgradeCcm(String environmentCrn) {
+        String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
```

Comment:
I would add a debug or info log about the triggering of CCM upgrade with the related env's crn.

Reply:
ok